interface/src/app/tribe_client/genesets.js

```diff
@@ -0,0 +1,97 @@
+angular.module('adage.tribe_client.genesets', [
+  'adage.tribe_client.resource'
+])
+
+// Geneset Search Factory
+// Allows for retreiving for and interacting with search results for genesets.
+.factory('GenesetSearch', ['Genesets', function(Genesets) {
+  var genesets = [];
+  var query = {};
+  var resultCount = null;
+
+  return {
+    getQuery: function() {
+      return query;
+    },
+    getGenesets: function() {
+      return genesets;
+    },
+    clear: function() {
+      query = {};
+      genesets = [];
+    },
+    getResultCount: function() {
+      return resultCount;
+    },
+    query: function(searchParams) {
+      query = searchParams;
+      Genesets.query(query, function(data) {
+        resultCount = data.meta.total_count;
+        genesets = data.objects;
+      });
+    }
+  };
+}])
+
+.directive('genesetSearchForm', ['GenesetSearch', function(GenesetSearch) {
+  return {
+    controller: ['$scope', function($scope) {
+      // Number of results (in this case genesets) to come back
+      // in each response page.
+      $scope.limit = 10;
+    }],
+    replace: true,
+    restrict: 'E',
+    scope: {
+      limit: '=',
+      organism: '='
+    },
+    templateUrl: 'tribe_client/geneset-search-form.tpl.html'
+  };
+}])
+
+
+// Search box for users to enter their search text into
+// and go search Tribe genesets.
+.directive('genesetSearchBar', ['GenesetSearch', function(GenesetSearch) {
+  return {
+    controller: ['$scope', function($scope) {
+      $scope.search = {};
+      $scope.search['organism__scientific_name'] = $scope.organism;
+      $scope.search['limit'] = $scope.limit;
+
+      $scope.searchGenesets = function() {
+        GenesetSearch.query($scope.search);
```

Comment:
Ahh, I see what is going on @mhuyck. It looks like the $scope.search is getting set correctly, since it is getting it from the same template it is bound to, not the parent one (I think I should definitely remove the ``search`` parameter from the searchGenesets() call in the template though). I think what is happening though is that the genesets aren't being returned by the query until *after* I call the GenesetSearch.getGenesets() (therefore, at the time when I do, this still returns an empty list). I should wait for the query to finish, and *then * ask to get the genesets back from the factory. I'll check to see if this approach works.